plugins/suggested-blocks/test/suggested_blocks.mocha.js

```diff
@@ -0,0 +1,229 @@
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'colour_picker',
+    'colour_picker',
+    'colour_random',
+    'colour_random',
+    'colour_random',
+  ],
+  expectedFrequentBlocks: [
+    'controls_if',
+    'controls_whileUntil',
+    'colour_random',
+    'colour_picker',
+    'lists_length',
+  ],
+  expectedRecentBlocks: [
+    'colour_random',
+    'colour_picker',
+    'controls_whileUntil',
+    'lists_length',
+    'controls_if',
+  ],
+};
+
+const MANY_UNIQUE_BLOCKS_TEST_CASE = {
+  usedBlockList: [
+    'controls_if',
+    'controls_if',
+    'lists_length',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'controls_whileUntil',
+    'colour_picker',
+    'colour_picker',
+    'colour_random',
+    'colour_random',
+    'math_arithmetic',
+    'logic_operation',
+    'math_single',
+    'math_trig',
+    'math_constant',
+    'controls_repeat_ext',
+    'controls_forEach',
+    'text_multiline',
+  ],
+  expectedFrequentBlocks: [
+    'controls_whileUntil',
+    'colour_random',
+    'colour_picker',
+    'controls_if',
+    'text_multiline',
+    'controls_forEach',
+    'controls_repeat_ext',
+    'math_constant',
+    'math_trig',
+    'math_single',
+
+  ],
+  expectedRecentBlocks: [
+    'text_multiline',
+    'controls_forEach',
+    'controls_repeat_ext',
+    'math_constant',
+    'math_trig',
+    'math_single',
+    'logic_operation',
+    'math_arithmetic',
+    'colour_random',
+    'colour_picker',
+  ],
+};
+
+const FREQUENCY_TIEBREAK_TEST_CASE = {
+  usedBlockList: [
+    // Used once
+    'controls_if',
+    'lists_length',
+    'controls_whileUntil',
+    'colour_picker',
+    'colour_random',
+    'math_arithmetic',
+    // Used a second time
+    'controls_if',
+    'controls_whileUntil',
+    'colour_picker',
+    'colour_random',
+    // Used a third time
+    'colour_picker',
+    'colour_random',
+  ],
+  expectedFrequentBlocks: [
+    'colour_random',
+    'colour_picker',
+    'controls_whileUntil',
+    'controls_if',
+    'math_arithmetic',
+    'lists_length',
+  ],
+  expectedRecentBlocks: [
+    'colour_random',
+    'colour_picker',
+    'controls_whileUntil',
+    'controls_if',
+    'math_arithmetic',
+    'lists_length',
+  ],
+};
+
+
+suite('Suggested blocks', function() {
+  /**
+   * Asserts that the list of blocks matches the expected list.
+   * @param {Array <object>} blockList list of block JSON objects
+   * @param {Array <string>} expectedBlockIds list of expected block_type values
+   */
+  function assertSuggestedListEquals(blockList, expectedBlockIds) {
+    const actualBlockIds = blockList.map((x) => x.type);
+    assert.deepEqual(actualBlockIds, expectedBlockIds);
+  }
+
+  const simulateTestCase = (testCase, workspace, clock) => {
+    for (const blockType of testCase.usedBlockList) {
+      workspace.newBlock(blockType);
+    }
+
+    // Wait 10 ms for the async BLOCK_CREATE events to propagate.
```

Comment:
afaik `clock.tick` doesn't make the thread wait, it actually forces timers to be fired.
```suggestion
    // Force events to fire by running any timers scheduled for the next 10ms.
```

Reply:
Updated.